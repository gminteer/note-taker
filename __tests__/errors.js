const {NotesError} = require('../lib/errors');
describe('NotesError', () => {
  test("shouldn't replace message if one is provided to contructor", () => {
    const testError = new NotesError(null, null, 'test message');
    expect(testError.message).toEqual('test message');
  });
  test('should set a message based on provided error code if no message is provided', () => {
    const notFoundError = new NotesError('NOTE_NOT_FOUND', {id: 'test'});
    expect(notFoundError).toMatchSnapshot();
    const failedValidationError = new NotesError('NOTE_FAILED_VALIDATION');
    expect(failedValidationError).toMatchSnapshot();
    const writeError = new NotesError('NOTE_WRITE_ERROR', {code: 'test'});
    expect(writeError).toMatchSnapshot();
  });
});
