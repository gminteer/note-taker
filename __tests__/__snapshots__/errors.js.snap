// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesError should set a message based on provided error code if no message is provided 1`] = `[Error: "test" not in notes array]`;

exports[`NotesError should set a message based on provided error code if no message is provided 2`] = `[Error: suggested note does not validate]`;

exports[`NotesError should set a message based on provided error code if no message is provided 3`] = `[Error: write failed with "test"]`;
